src/owr/OptimisticWithdrawalRecipientPectraFactory.sol

```diff
@@ -0,0 +1,119 @@
+
+  constructor(string memory _ensName, address _ensReverseRegistrar, address _ensOwner, address _pectraWithdrawalAddress, address _pectraConsolidationAddress) {
+    owrImpl = new OptimisticWithdrawalRecipientPectra(_pectraWithdrawalAddress, _pectraConsolidationAddress);
+    owrImpl.initialize(address(this));
+    IENSReverseRegistrar(_ensReverseRegistrar).setName(_ensName);
+    IENSReverseRegistrar(_ensReverseRegistrar).claim(_ensOwner);
+  }
+
+  /// -----------------------------------------------------------------------
+  /// functions
+  /// -----------------------------------------------------------------------
+
+  /// -----------------------------------------------------------------------
+  /// functions - public & external
+  /// -----------------------------------------------------------------------
+
+  /// Create a new OptimisticWithdrawalRecipientPectra clone
+  /// @param recoveryAddress Address to recover tokens to
+  /// If this address is 0x0, recovery of unrelated tokens can be completed by
+  /// either the principal or reward recipients.  If this address is set, only
+  /// this address can recover
+  /// tokens (or ether) that isn't the token of the OWRecipient contract
+  /// @param principalRecipient Address to distribute principal payments to
+  /// @param rewardRecipient Address to distribute reward payments to
+  /// @param amountOfPrincipalStake Absolute amount of stake to be paid to
+  /// principal recipient (multiple of 32 ETH)
+  /// (reward recipient has no threshold & receives all residual flows)
+  /// it cannot be greater than uint96
+  /// @return owr Address of new OptimisticWithdrawalRecipientPectra clone
+  function createOWRecipient(
+    address recoveryAddress,
+    address principalRecipient,
+    address rewardRecipient,
+    uint256 amountOfPrincipalStake,
+    address owner
+  ) external returns (OptimisticWithdrawalRecipientPectra owr) {
+    /// checks
+
+    // ensure doesn't have address(0)
+    if (principalRecipient == address(0) || rewardRecipient == address(0)) revert Invalid__Recipients();
+    // ensure threshold isn't zero
+    if (amountOfPrincipalStake == 0) revert Invalid__ZeroThreshold();
+    // ensure threshold isn't too large
+    if (amountOfPrincipalStake > type(uint96).max) revert Invalid__ThresholdTooLarge(amountOfPrincipalStake);
+
+    /// effects
+    uint256 principalData = (amountOfPrincipalStake << ADDRESS_BITS) | uint256(uint160(principalRecipient));
+    uint256 rewardData = uint256(uint160(rewardRecipient));
+
+    // would not exceed contract size limits
+    // important to not reorder
+    bytes memory data = abi.encodePacked(recoveryAddress, principalData, rewardData);
+    owr = OptimisticWithdrawalRecipientPectra(address(owrImpl).clone(data));
+    owr.initialize(owner);
+
+    emit CreateOWRecipient(address(owr), recoveryAddress, principalRecipient, rewardRecipient, amountOfPrincipalStake);
```

Comment:
Should we log the owner? (how many things can we include in a log? i thought it was 4, 3 if named, but this has 5 and I'm suggesting a 6th

Reply:
You can include more than 3. You can only have 3 index parameters, but you can add more non-indexed ones. Added the owner as well